### PR TITLE
Correction Move T4 delta/count_prev to callback and Add Missing End Function

### PR DESCRIPTION
**Pull request by mjs513**
This adds in the missing FreqCount.end() function identified in https://forum.pjrc.com/threads/58014-Teensy-4-FreqCount-compile-error

### CODE REVIEW

**Comment by mjs513** [comment]
@PaulStoffregen - @joepasquariello

Updated this PR to include the corrections to move delta = count - prev and prev = count into the timer callback function in FreqCountTimers.h.  The discussion of the changes start at https://forum.pjrc.com/threads/68386-Teensyduino-1-56-Beta-2?p=292707&viewfull=1#post292707.  @joepasquariello was able to test the changes using his freq generator up to 3.1Mhz: https://forum.pjrc.com/threads/68386-Teensyduino-1-56-Beta-2?p=292722&viewfull=1#post292722

File: FreqCount.cpp [hunk 1]
```diff
@@ -126,15 +126,17 @@ uint8_t FreqCountClass::available(void)
 
 uint32_t FreqCountClass::read(void)
 {
-    count_output = count - count_prev;
-    count_prev = count;
-    //Serial.println(count - count_prev);
-    count_prev = count;
     count_ready = 0;
+	count = 0;
```
Comment (ssilverman) [inline FreqCount.cpp:130]: Spacing looks off. Tabs vs. spaces?

**Comment by joepasquariello** [comment]
I did my testing without the added line "count = 0" in read(), which I think should be removed. As long as read() is not called without first checking available(), there is no issue. However, if read() is called without calling first checking available(), having count = 0 in read() will result in some calls returning 0, and other calls returning a real measurement, which sounds like what you were seeing in your tests. If read() does not set count = 0, calls to read() will always return the last actual measurement, and that seems more logical to me.

File: FreqCount.cpp (hunk 1 above)
Comment (mjs513) [reply FreqCount.cpp]: Fixed spacing and pushed changes.

**Comment by mjs513** [comment]
Did some retesting right now with count = 0 removed and now I am not seeing that alternating 0 and 50mhz counts.  This was tested on a unlocked T4.0.

Since I can not duplicate the issue I was seeing I am goin going to remove count = 0 from the read() function.  Even if I removed FreqCount.available it is still working.

**Pull request merged**